webpack.conf.js

```diff
@@ -1,36 +1,80 @@
+    LodashModuleReplacementPlugin = require('lodash-webpack-plugin'),
+    path = require('path'),
+    webpack = require('webpack');
 
 module.exports = {
-    watch: false,
-    devtool: 'source-map',
+    bail: true,
     context: __dirname,
+    devtool: 'source-map',
     entry: {
-        'theme-bundle': './assets/js/app.js',
-    },
-    output: {
-        path: `./assets/dist`,
-        filename: '[name].js',
+        main: './assets/js/app.js',
     },
-    bail: true,
     module: {
-        loaders: [
+        rules: [
             {
                 test: /\.js$/,
-                loader: 'babel',
                 include: /(assets\/js|assets\\js|stencil-utils)/,
-                query: {
-                    compact: false,
-                    cacheDirectory: true,
-                    presets: ['es2015-loose'],
+                use: {
+                    loader: 'babel-loader',
+                    options: {
+                        cacheDirectory: true,
+                        compact: true,
+                        minified: true,
+                        plugins: [
+                            'dynamic-import-webpack', // Needed for dynamic imports.
+                            'lodash', // Automagically tree-shakes lodash.
+                            'transform-regenerator', // Transforms async and generator functions.
+                        ],
+                        presets: [
+                            ['env', {
+                                loose: true, // Enable "loose" transformations for any plugins in this preset that allow them.
+                                modules: false, // Don't transform modules; needed for tree-shaking.
+                                useBuiltIns: true, // Tree-shake babel-polyfill.
+                            }],
+                        ],
+                    }
                 }
             }
-        ]
+        ],
+    },
+    output: {
+        chunkFilename: 'theme-bundle.chunk.[name].js',
+        filename: 'theme-bundle.[name].js',
+        path: path.resolve(__dirname, "assets/dist"),
     },
     plugins: [
+        new CleanWebpackPlugin(['assets/dist'], {
+            verbose: true,
+            watch: false,
+        }),
+        new webpack.LoaderOptionsPlugin({
+             minimize: true,
+        }),
+        new LodashModuleReplacementPlugin, // Complements 'babel-plugin-lodash by shrinking it's cherry-picked builds further.
         new webpack.ProvidePlugin({
             $: 'jquery',
             jQuery: 'jquery',
             'window.jQuery': 'jquery',
-        })
+        }),
+        new webpack.optimize.CommonsChunkPlugin({
+            name: 'vendor',
+            minChunks: function (module) {
+                // this assumes your vendor imports exist in the node_modules directory
+                return module.context && module.context.indexOf('node_modules') !== -1;
+            },
+        }),
+        new webpack.optimize.CommonsChunkPlugin({
```

Comment:
this one pulls duplicate child files out of the code split chunks into their parent